app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -28,7 +29,14 @@ class Api::V2::SubscriptionsController < Api::V2::ApiController
     param :match_installed, :bool, :desc => N_("Return subscriptions that match installed products of the specified host")
     param :no_overlap, :bool, :desc => N_("Return subscriptions which do not overlap with a currently-attached subscription")
     def index
-      respond(:collection => scoped_search(index_relation.uniq, :cp_id, :asc, :resource_class => Pool, :includes => [:subscription]))
+      collection = scoped_search(
+        index_relation.uniq, :cp_id, :asc, resource_class: Pool, includes: [:subscription])
+      if params[:activation_key_id]
+        collection[:results] = collection[:results].map do |pool|
+          pool.with_quantity_attached(@activation_key)
+        end
+      end
+      respond(:collection => collection)
     end
```

Comment:
@ehelms @jlsherrill I dont think it is an issue, but just want to confirm that it is ok to modify the results for an index call like this?